src/generators/default_map_generator_job.cpp

```diff
@@ -628,152 +707,53 @@ static void flood_name(const map_location& start, const std::string& name, std::
 	}
 }
 
-namespace {
-
-// the configuration file should contain a number of [height] tags:
-//   [height]
-//     height=n
-//     terrain=x
-//   [/height]
-// These should be in descending order of n.
-// They are checked sequentially, and if height is greater than n for that tile,
-// then the tile is set to terrain type x.
-class terrain_height_mapper
-{
-public:
-	explicit terrain_height_mapper(const config& cfg);
-
-	bool convert_terrain(const int height) const;
-	t_translation::t_terrain convert_to() const;
-
-private:
-	int terrain_height;
-	t_translation::t_terrain to;
-};
-
-terrain_height_mapper::terrain_height_mapper(const config& cfg) :
-	terrain_height(cfg["height"]),
-	to(t_translation::GRASS_LAND)
-{
-	const std::string& terrain = cfg["terrain"];
-	if(terrain != "") {
-		to = t_translation::read_terrain_code(terrain);
-	}
-}
-
-bool terrain_height_mapper::convert_terrain(const int height) const
-{
-	return height >= terrain_height;
-}
-
-t_translation::t_terrain terrain_height_mapper::convert_to() const
-{
-	return to;
-}
-
-class terrain_converter
-{
-public:
-	explicit terrain_converter(const config& cfg);
-
-	bool convert_terrain(const t_translation::t_terrain & terrain, const int height, const int temperature) const;
-	t_translation::t_terrain convert_to() const;
-
-private:
-	int min_temp, max_temp, min_height, max_height;
-	t_translation::t_list from;
-	t_translation::t_terrain to;
-};
-
-terrain_converter::terrain_converter(const config& cfg)
-	: min_temp(cfg["min_temperature"].to_int(-100000))
-	, max_temp(cfg["max_temperature"].to_int(100000))
-	, min_height(cfg["min_height"].to_int(-100000))
-	, max_height(cfg["max_height"].to_int(100000))
-	, from(t_translation::read_list(cfg["from"]))
-	, to(t_translation::NONE_TERRAIN)
-{
-	const std::string& to_str = cfg["to"];
-	if(to_str != "") {
-		to = t_translation::read_terrain_code(to_str);
-	}
-}
-
-bool terrain_converter::convert_terrain(const t_translation::t_terrain & terrain,
-		const int height, const int temperature) const
-{
-	return std::find(from.begin(),from.end(),terrain) != from.end() && height >= min_height && height <= max_height && temperature >= min_temp && temperature <= max_temp && to != t_translation::NONE_TERRAIN;
-}
-
-t_translation::t_terrain terrain_converter::convert_to() const
-{
-	return to;
-}
-
-} // end anon namespace
-
-std::string default_map_generator_job::default_generate_map(size_t width, size_t height, size_t island_size, size_t island_off_center, size_t iterations, size_t hill_size, size_t max_lakes, size_t nvillages, size_t castle_size, size_t nplayers, bool roads_between_castles,
-								 std::map<map_location,std::string>* labels, const config& cfg)
+std::string default_map_generator_job::default_generate_map(
```

Comment:
I honestly don't think it really helps much to split ridiculously long function headers into multiple lines, but whatever...